### PR TITLE
Add sigv4 module

### PR DESCRIPTION
**Pull request by treid314**
This contains the sigv4 config and sig v4 roundtripper. Do I need to add a new makefile for this module as well?

### CODE REVIEW

File: sigv4/sigv4.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+// newSigV4RoundTripper returns a new http.RoundTripper that will sign requests
+// using Amazon's Signature Verification V4 signing procedure. The request will
+// then be handed off to the next RoundTripper provided by next. If next is nil,
+// http.DefaultTransport will be used.
+//
+// Credentials for signing are retrieved using the the default AWS credential
+// chain. If credentials cannot be found, an error will be returned.
+func newSigV4RoundTripper(cfg *SigV4Config, next http.RoundTripper) (http.RoundTripper, error) {
```
Comment (pracucci) [inline sigv4/sigv4.go:52]: This function is not exported. I guess you want to export it.

```suggestion
func NewSigV4RoundTripper(cfg *SigV4Config, next http.RoundTripper) (http.RoundTripper, error) {
```

**Comment by roidelapluie** [comment]
yes, it would be nice to add a Makefile and call it from the main makefile

File: sigv4/sigv4_config.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	Profile   string        `yaml:"profile,omitempty"`
+	RoleARN   string        `yaml:"role_arn,omitempty"`
+}
+
+func (c *SigV4Config) Validate() error {
+	if (c.AccessKey == "") != (c.SecretKey == "") {
+		return fmt.Errorf("must provide a AWS SigV4 Access key and Secret Key if credentials are specified in the SigV4 config")
+	}
+	return nil
+}
```
Comment (roidelapluie) [inline sigv4/sigv4_config.go:38]: Missing empty line.

**Comment by roidelapluie** [comment]
Thanks, I have added 2 small comments.

File: .circleci/config.yml [hunk 3]
```diff
@@ -35,6 +35,9 @@ jobs:
         steps:
         - go/save-cache:
             key: v1-go<< parameters.go_version >>
+    - run:
+        command: make test
```
Comment (pracucci) [inline .circleci/config.yml:39]: What's about `make style`? I think we should add a run step for sigv4 above, under:
```
    - when:
        condition: << parameters.run_style >>
        steps:
        - run: make style
```

File: sigv4/Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include ../Makefile.common
+
+
```
Comment (pracucci) [inline sigv4/Makefile:16]: [nit] Extra new line to be removed

File: .circleci/config.yml (hunk 3 above)
Comment (roidelapluie) [reply .circleci/config.yml:39]: +1

**Comment by roidelapluie** [comment]
Thanks, we are getting close :)

**Review by pracucci — APPROVED** [review]
LGTM!

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**